### PR TITLE
[Feature]Add config table

### LINKED ISSUE #1630: [feature] system.configs table
**Summary**

Add system.configs table to show the configs:
columns:
`name | value | group | description`

```
 select * from system.configs;

┌─name──────────────────┬─value─────┬─group─────────┬─description──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│ num_cpus              │ 10        │ Query         │ Maximum thread to use for the query                                                                                                  │
└───────────────────────┴───────────┴───────────────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

Reference:
system.settings table:
https://github.com/datafuselabs/datafuse/blob/master/query/src/datasources/system/settings_table.rs

### PR DESCRIPTION
**Pull request by GrapeBaBa**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

Add config table

## Changelog

- New Feature

## Related Issues

Fixes #1630 

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: query/src/datasources/system/configs_table_test.rs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        ctx.get_settings().get_max_threads()? as usize,
+    )?;
+
+    let stream = table.read(ctx, &source_plan).await?;
+    let result = stream.try_collect::<Vec<_>>().await?;
+    let block = &result[0];
+    assert_eq!(block.num_columns(), 4);
+    assert_eq!(block.num_rows(), 30);
+
+    let expected = vec![
```
Comment (sundy-li) [inline query/src/datasources/system/configs_table_test.rs:50]: What if we add one config, the tests will be failed. So I suggest just to test some common values?

Comment (GrapeBaBa) [reply query/src/datasources/system/configs_table_test.rs:50]: any method can be invoked to get rows on result?

Comment (BohuTANG) [reply query/src/datasources/system/configs_table_test.rs:50]: Like some column names :)

**Comment by GrapeBaBa** [comment]
@BohuTANG @sundy-li I will change the test later when I am available.

**Pull request merged**